nmigen_cocotb.py

```diff
@@ -85,6 +123,9 @@ def run(design, module, platform=None, ports=(), name='top', verilog_sources=Non
             sources.extend(verilog_sources)
         if extra_files:
             copy_extra_files(extra_files, d)
+        if platform and platform.extra_files:
+            plat_extra = platform.dump_extra_files(d)
+            sources.extend(plat_extra)
```

Comment:
The extra_files can be a memory dumps. tcl scripts or any other file. We should distinguish if it is a verilog or system verilog file.

Reply:
you are totally right! What about now?

Reply:
done